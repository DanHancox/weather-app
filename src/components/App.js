import React, { useState, useEffect } from "react";

import "../styles/App.css";
import getForecast from "../requests/getForecast";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import SearchForm from "./SearchForm";

function App() {
  const [selectedDate, setSelectedDate] = useState(0);
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });

  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );

  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };

  useEffect(() => {
    getForecast(setSelectedDate, setForecasts, setLocation);
  }, []);

  return (
    <div className="weather-app">
      <LocationDetails city={location.city} country={location.country} />

      <SearchForm />

      <ForecastSummaries
        forecasts={forecasts}
        onForecastSelect={handleForecastSelect}
      />

      {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
    </div>
  );
}

export default App;
