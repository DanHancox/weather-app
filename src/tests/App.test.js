import { render, screen } from '@testing-library/react';
import App from '../components/App';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/Weather app/i);
  expect(linkElement).toBeInTheDocument();
});
