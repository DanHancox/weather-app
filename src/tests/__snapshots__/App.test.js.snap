// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1>
      , 
    </h1>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;

exports[`App renders App as asserted 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1>
      , 
    </h1>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
